Fix/wifi 11k roaming fail

fengming-ye opened the pull request:
233b6e8b404c6ff7f8bc8a4834ac38b40146b849
subsys: wifi: fix wifi 11k cannot be set before connection
    
    Support the case that 11k parameter can be set before wifi connection.
    This field should not be cleared by connection.
    Fix it by not clear it in wifi_connect api.

80d1d234965ca3429e6e1c5629f6bfaf1dab78a2
modules: hostap: fix wifi roaming aborted in few cases
    
    Add guard of supplicant state condition.
    Not send neighbor request in auth procedures.

fengming-ye committed:
net: wifi: fix 11k cannot be set before connection

Support the case that 11k parameter can be set before wifi connection.
This field should not be cleared by connection.
Neighbor report will be cleared before start_roaming.
So wifi_connect don't need to clear 11k params.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

krish2718 commented on modules/hostap/src/supp_api.c:
```diff
@@ -1528,12 +1528,25 @@ int supplicant_11k_cfg(const struct device *dev, struct wifi_11k_params *params)
+		return -1;
+	}
+
+	if (wpa_s->reassociate || (wpa_s->wpa_state >= WPA_AUTHENTICATING &&
```
can't we just do below?

```
if (wpa_s->wpa_stats != WPA_COMPLETED) {
 return;
}
```

fengming-ye replied:
just copy same guard as supplicant_11r_roaming here.
I don't want to block more roaming actions than which during auth, in case some of them are expected.

fengming-ye replied:
most cases yes. But can we send 11k while WPA_SCANNING? I'm not sure. So I just copy 11r roaming entry condition here.

krish2718 commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -442,7 +442,7 @@ static int wifi_connect(uint64_t mgmt_request, struct net_if *iface,
 
 #ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_ROAMING
-	memset(&roaming_params, 0x0, sizeof(roaming_params));
+	memset((void *)&roaming_params.neighbor_rep, 0x0, sizeof(roaming_params.neighbor_rep));
```
Can you please rewrite the commit log? IIUC, you are saying that `Roaming state should not be cleared for every connection request, only 11k state needs to be cleared"?

fengming-ye replied:
yes sure

fengming-ye committed:
modules: hostap: fix wifi roaming aborted in few cases

Add guard of supplicant state condition.
Not send neighbor request in auth procedures.
Add process for null pointer params.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye commented:
reworded commit log to fix compliance check
